src/app/activity/bounty/Claim.ts

```diff
@@ -68,56 +100,147 @@ const getDbHandler = async (request: ClaimRequest): Promise<{dbBountyResult: Bou
+            },
+        },
+    });
 
     if (writeResult.result.ok !== 1) {
-        Log.error(`Write result did not execute correctly`);
+        Log.error('failed to update claimed bounty with in progress status');
         throw new Error(`Write to database for bounty ${request.bountyId} failed for ${request.activity} `);
     }
+
+    return claimedBounty;
 }
 
-export const claimBountyMessage = async (message: Message, claimedByUser: GuildMember): Promise<any> => {
-	Log.debug(`fetching bounty message for claim`)
+export const claimBountyMessage = async (message: Message, claimedBounty: BountyCollection, claimedByUser: GuildMember, originalBounty: BountyCollection): Promise<any> => {
+    Log.debug(`fetching bounty message for claim`)
     
-    const embedMessage: MessageEmbed = message.embeds[0];
+    const existingEmbeds = message.embeds[0];
+    const embedNewMessage: MessageEmbed = new MessageEmbed(existingEmbeds);
 
-	embedMessage.fields[BountyEmbedFields.status].value = BountyStatus.in_progress;
-	embedMessage.setColor('#d39e00');
-	embedMessage.addField('Claimed by', claimedByUser.user.tag, true);
+    // Send claimed bounty by DM
+    embedNewMessage.fields[BountyEmbedFields.status].value = BountyStatus.in_progress;
+    embedNewMessage.fields[BountyEmbedFields.bountyId].value = claimedBounty._id.toString();
+    embedNewMessage.setTitle(BountyUtils.createPublicTitle(<Bounty>claimedBounty));
+    embedNewMessage.setURL(process.env.BOUNTY_BOARD_URL + claimedBounty._id.toString());
+    embedNewMessage.setColor('#d39e00');
+    embedNewMessage.addField('Claimed by', claimedByUser.user.tag, true);
+    embedNewMessage.setFooter({text: '📮 - submit | 🆘 - help'});
+    const claimantMessage: Message = await claimedByUser.send({ embeds: [embedNewMessage] });
+    await addClaimReactions(claimantMessage);
 
-	embedMessage.setFooter('📮 - submit | 🆘 - help');
-	await message.edit({ embeds: [embedMessage] });
-	await addClaimReactions(message);
+    // If Bounty status is no longer open, delete the board message, otherwise update title
+    if (originalBounty.status !== BountyStatus.open) {
+        await message.delete();
+    } else {
+        const embedOrigMessage: MessageEmbed = message.embeds[0];
+        embedOrigMessage.setTitle(BountyUtils.createPublicTitle(<Bounty>originalBounty));
+        await message.edit({ embeds: [embedOrigMessage] });
+    }
+    await updateMessageStore(claimedBounty, claimantMessage);
 };
 
 export const addClaimReactions = async (message: Message): Promise<any> => {
-	await message.reactions.removeAll();
-	await message.react('📮');
-	await message.react('🆘');
-};
\ No newline at end of file
+    // await message.reactions.removeAll();
+    await message.react('📮');
+    await message.react('🆘');
+};
+
+// Save where we sent the Bounty message embeds for future updates
+export const updateMessageStore = async (bounty: BountyCollection, message: Message): Promise<any> => {
```

Comment:
nice :)

Reply:
There are 2 other repeats of similar code. Should probably be refactored into Utils or something.